x-pack/plugins/lens/public/editor_frame_service/editor_frame/workspace_panel/workspace_panel.tsx

```diff
@@ -232,31 +232,47 @@ export const InnerWorkspacePanel = React.memo(function InnerWorkspacePanel({
       if (renderDeps.current) {
         const [defaultLayerId] = Object.keys(renderDeps.current.datasourceLayers);
 
-        const requestWarnings: string[] = [];
-        const datasource = Object.values(renderDeps.current?.datasourceMap)[0];
-        const datasourceState = Object.values(renderDeps.current?.datasourceStates)[0].state;
-        if (adapters?.requests) {
-          plugins.data.search.showWarnings(adapters.requests, (warning) => {
-            const warningMessage = datasource.getSearchWarningMessages?.(datasourceState, warning);
+        const warningsMap: Map<string, Array<string | React.ReactNode>> = new Map();
```

Comment:
Does it make sense to unify all this and the logic in the Embeddable?

For example, we could create a `getSearchWarningsCallback` function that would encapsulate all this stuff including the `Map` deduplication logic.

```ts
plugins.data.search.showWarnings(adapters.requests, getSearchWarningsCallback(datasource));
```

Reply:
Yes, we discussed this with @flash1293. But the problem is that we do not process all warnings in `Lens` and some of them are still processed in the `data` plugin.

The current API does not allow you to do as you suggest. Please see `src/plugins/data/public/search/fetch/handle_warnings.tsx` file.